packages/rs-drive/src/drive/identity/update/operations/merge_identity_nonce_operations/v0/mod.rs

```diff
@@ -0,0 +1,156 @@
+use crate::drive::identity::{identity_path, identity_path_vec, IdentityRootStructure};
+use crate::drive::Drive;
+
+use crate::error::Error;
+
+use crate::fee::op::LowLevelDriveOperation;
+use grovedb::batch::KeyInfoPath;
+
+use dpp::prelude::IdentityNonce;
+
+use dpp::version::PlatformVersion;
+use grovedb::{Element, EstimatedLayerInformation, TransactionArg};
+
+use crate::drive::identity::IdentityRootStructure::IdentityTreeNonce;
+use crate::drive::object_size_info::PathKeyElementInfo;
+use crate::error::identity::IdentityError;
+use dpp::block::block_info::BlockInfo;
+use dpp::identity::identity_nonce::MergeIdentityNonceResult::{
+    MergeIdentityNonceSuccess, NonceAlreadyPresentAtTip, NonceAlreadyPresentInPast,
+    NonceTooFarInFuture, NonceTooFarInPast,
+};
+use dpp::identity::identity_nonce::{
+    MergeIdentityNonceResult, IDENTITY_NONCE_VALUE_FILTER, IDENTITY_NONCE_VALUE_FILTER_MAX_BYTES,
+    MAX_MISSING_IDENTITY_REVISIONS, MISSING_IDENTITY_REVISIONS_FILTER,
+    MISSING_IDENTITY_REVISIONS_MAX_BYTES,
+};
+use std::collections::HashMap;
+
+impl Drive {
+    /// Update the nonce of the identity
+    /// Nonces get bumped on all identity state transitions except those that use an asset lock
+    pub(in crate::drive::identity::update) fn merge_identity_nonce_operations_v0(
```

Comment:
Review this

Reply:
The same. I recommend to write a unit test for this function since it contains a lot of logic